app/__init__.py

```diff
@@ -233,6 +233,21 @@ def create_app(application):
     )
     logging.init_app(application)
 
+    # Hopefully will help identify if there is a race condition causing the CSRF errors
+    # that we have occasionally seen in our environments.
+    for key in ("SECRET_KEY", "DANGEROUS_SALT"):
+        try:
+            value = application.config[key]
+        except KeyError:
+            application.logger.error(f"Env Var {key} doesn't exist.")
+        else:
+            try:
+                data_len = len(value.strip())
+            except (TypeError, AttributeError):
+                application.logger.error(f"Env Var {key} invalid type: {type(value)}")
+            else:
+                application.logger.info(f"Env Var {key} Length: {data_len}")
```

Comment:
Okay, this should hopefully help determine if this is what we're facing, thank you!

One last security observation:  instead of displaying the length of the key itself, what if we just indicate whether or not it's empty?  Do you think that would suffice?  e.g.:

```suggestion
                if data_len > 0:
                    application.logger.info(f"Env Var {key} is a non-zero length.")
                else:
                    application.logger.info(f"Env Var {key} is empty.")
```

My concern is that by logging the actual length of the key and if that were to somehow be exposed, that gives a bit of info to attackers on how long the key's string length is, thus reducing the amount of possible things to try.